app/filtering.js

```diff
@@ -761,46 +761,55 @@ module.exports.initPartition = initPartition
 
 function setupTor () {
   let torInitialized = null
+  const setTorErrorOnTimeout = (timeout, msg) => {
+    torInitialized = null
+    setTimeout(() => {
+      if (torInitialized === null) {
+        appActions.onTorInitError(msg)
+      }
+    }, timeout)
+  }
+  const onTorSuccess = () => {
+    torInitialized = true
+    appActions.onTorInitSuccess()
+  }
+  const onTorFail = (msg) => {
+    torInitialized = false
+    appActions.onTorInitError(msg)
```

Comment:
@riastradh-brave onTorInitError shouldn't be called twice because https://github.com/brave/browser-laptop/pull/14499/files#diff-8a9bac13ef6264feb2b6da7c18d86b3cR767 (torInitialized is set to false by the time the timeout happens). either way it's harmless if it gets called twice in a row - only one dialog will ever appear at a time.

i think you're right that we only need 2 states though - one for tor status determined and one for tor status undetermined. the timeout currently doesn't care about true vs false.